### PR TITLE
fix(nx-plugin): resolve new swc transpilation regression within build executor

### PR DESCRIPTION
**Pull request by Hotell**
## New behaviour:

- build executor uses new swc cli argument https://swc.rs/docs/usage/cli#--strip-leading-paths in order to mitiage regression that will create build assets within provided source root

- force generated change file for resize-handler in order to trigger new release for affected package
- added tests for build executor

---

- Fixes https://github.com/microsoft/fluentui-contrib/issues/198

### CODE REVIEW

File: packages/nx-plugin/package.json [hunk 1]
```diff
@@ -6,10 +6,10 @@
   "generators": "./generators.json",
   "executors": "./executors.json",
   "dependencies": {
+    "@nx/devkit": "^19.3.2",
     "semver": "^7.5.2",
-    "@nx/devkit": "^17.3.0",
```
Comment (Hotell) [inline packages/nx-plugin/package.json:10]: leftover from https://github.com/microsoft/fluentui-contrib/pull/191

File: packages/nx-plugin/src/executors/build/executor.ts [hunk 2]
```diff
@@ -12,20 +18,21 @@ export default async function runExecutor(
...
+    );
   }
 
-  const { sourceRoot, root } =
+  const projectConfig =
     context.projectsConfigurations.projects[context.projectName];
 
-  const { root: workspaceRoot } = context;
-  const paths = getPackagePaths(workspaceRoot, root);
+  if (!projectConfig) {
```
Comment (Hotell) [inline packages/nx-plugin/src/executors/build/executor.ts:29]: changed validation logic - we care only if project exists or not

File: packages/nx-plugin/src/executors/build/executor.ts [hunk 3]
```diff
@@ -42,10 +49,13 @@ function copyReadme(paths: PackagePaths) {
...
 }
 
+function cleanDist(paths: PackagePaths) {
+  fs.rmSync(paths.dist, { recursive: true, force: true });
+}
+
 function copyPackageJson(paths: PackagePaths) {
   const packageJson = readJsonFile(paths.packageJson);
   Object.assign(packageJson, {
-    types: './src/index.d.ts',
```
Comment (Hotell) [inline packages/nx-plugin/src/executors/build/executor.ts:48]: dupe for tsc d.ts type retrieval - unified with fluent core

File: packages/nx-plugin/src/utils.ts [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
-    main: path.join(rootPath, 'src', 'index.ts'),
-    src: path.join(rootPath, 'src'),
-    packageJson: path.join(rootPath, 'package.json'),
-    readme: path.join(rootPath, 'README.md'),
-    swcrc: path.join(rootPath, '.swcrc'),
-    tsconfigLib: path.join(rootPath, 'tsconfig.lib.json'),
-    dist: path.join(workspaceRoot, 'dist', rootPath),
-    esm: path.join(workspaceRoot, 'dist', rootPath, 'lib'),
-    commonjs: path.join(workspaceRoot, 'dist', rootPath, 'lib-commonjs'),
+    main: joinPathFragments(rootPath, 'src', 'index.ts'),
```
Comment (Hotell) [inline packages/nx-plugin/src/utils.ts:10]: less repetition - DRY

**Review by tudorpopams — APPROVED** [review]

**Review by mainframev — APPROVED** [review]

**Pull request merged**